src/moi.jl

```diff
@@ -226,55 +227,33 @@ function presolve!(dest::MOI.ModelLike, src::MOI.ModelLike, T::Type{<:Real})
                 scalar_set,
             )
         end
-        return MOI.OPTIMIZE_NOT_CALLED, x -> _not_inferred_postsolve_fn(ps, x)
     end
+    return model_status, x -> _postsolve_fn(ps, x)
 end
 
-function _optimal_postsolve_fn(ps::PresolveData{T}, x::Vector{T}) where {T}
-    if !isempty(x)
-        throw(ArgumentError("Presolve solved model to optimality; postsolve expects an empty input argument."))
-    end
-    @assert ps.solution.primal_status == FEASIBLE_POINT
-    @assert !ps.solution.is_primal_ray
-    @assert !ps.solution.is_dual_ray
-    return ps.solution.x
-end
-
-function _infeasible_postsolve_fn(ps::PresolveData{T}, x::Vector{T}) where {T}
-    if !isempty(x)
-        throw(ArgumentError("Presolve proved that model is infeasible; postsolve expects an empty input argument."))
-    end
-    @assert ps.solution.primal_status == NO_SOLUTION
-    @assert !ps.solution.is_primal_ray
-    if ps.solution.is_dual_ray
-        # TODO: Return dual ray here
-        return nothing
-    else
-        return nothing
-    end
-end
-
-function _unbounded_postsolve_fn(ps::PresolveData{T}, x::Vector{T}) where {T}
-    if !isempty(x)
-        throw(ArgumentError("Presolve proved that model is unbounded; postsolve expects an empty input argument."))
-    end
-    @assert ps.solution.dual_status == NO_SOLUTION
-    @assert !ps.solution.is_dual_ray
-    if ps.solution.is_primal_ray
+function _postsolve_fn(ps::PresolveData{T}, x::Vector{T}) where {T}
+    if ps.model_status == OPTIMAL
+        @assert ps.primal_status == FEASIBLE_POINT
+        if !isempty(x)
+            throw(ArgumentError("Presolve solved model to optimality; postsolve expects an empty input argument."))
+        end
+        return ps.solution.x
+    elseif ps.model_status == PRIMAL_INFEASIBLE
+        throw(ArgumentError("Presolve solved proven infeasible; cannot postsolve."))
+    elseif ps.model_status == DUAL_INFEASIBLE
+        @assert ps.is_primal_ray
+        # QUESTION: What do we know about ps.primal_status here?
```

Comment:
What status does Gurobi return for an unbounded MIP? I guess we can follow a similar convention?

Reply:
Dual infeasible: https://github.com/jump-dev/Gurobi.jl/blob/3e996da53396996ce40df4436eefccfa0a05b18a/src/MOI_wrapper.jl#L2437

Reply:
If presolve found the problem was unbounded, we should have found an unbounded ray too.
The rest should be identical to the `OPTIMAL` case, except that `ps.primal_status` should be `INFEASIBILITY_CERTIFICATE`.